storage/gcp/gcp.go

```diff
@@ -372,6 +428,7 @@ func (a *Appender) publishCheckpoint(ctx context.Context, size uint64, root []by
 type objStore interface {
 	getObject(ctx context.Context, obj string) ([]byte, int64, error)
 	setObject(ctx context.Context, obj string, data []byte, cond *gcs.Conditions, contType string, cacheCtl string) error
+	deleteObjectsWithPrefix(ctx context.Context, prefix string) error
```

Comment:
It's an internal interface, so I'm not overly worried, but this looks like a strange cousin next to these other 2 get/set methods.

Reply:
Open to alternative suggestions, but the thinking was that this enables infra to do this specific task in the best way it can (e.g. POSIX could potentially do an `rmdir()`) rather than force all to conform to an LCD like `findObjectsWithPrefix` and `deleteObject`